actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -33,6 +33,7 @@ def generate(name, options, path_parameters, parameterize = nil)
           defaults       = route.defaults
           required_parts = route.required_parts
           parameterized_parts.delete_if do |key, value|
+            next if defaults[key].nil?
```

Comment:
It becomes a bit long for my tastes, also requires a bang expression, which I try to avoid:

```
!defaults[key].nil? && value.to_s == defaults[key].to_s && !required_parts.include?(key)
```

You make the call. Which do you prefer?